lib/transaction/transaction.js

```diff
@@ -1371,6 +1370,8 @@ Transaction.prototype.seal = function () {
     self.inputs[key].setScript(unlockScript)
   })
 
+  this._updateChangeOutput()
```

Comment:
`this._updateChangeOutput()`.   after unlockingScript have been setted, at line `1370` 

```js
self.inputs[key].setScript(unlockScript)
```

this may cause signature  in the unlockingScript became invalid.